drizzlepac/hlautils/astrometric_utils.py

```diff
@@ -70,57 +68,36 @@
 
 MODULE_PATH = os.path.dirname(inspect.getfile(inspect.currentframe()))
 VEGASPEC = os.path.join(os.path.dirname(MODULE_PATH),
-                        'data','alpha_lyr_stis_008.fits')
+                        'data', 'alpha_lyr_stis_008.fits')
 
 __all__ = ['create_astrometric_catalog', 'compute_radius', 'find_gsc_offset',
-            'extract_sources', 'find_hist2d_offset', 'generate_source_catalog',
-            'classify_sources', 'countExtn']
+           'extract_sources', 'find_hist2d_offset', 'generate_source_catalog',
+           'classify_sources', 'countExtn']
 
-"""
-Utility functions.
 
-Many were originally released in stsci.tools.fileutil.
-"""
 def buildRotMatrix(theta):
     _theta = np.deg2rad(theta)
-    _mrot = np.zeros(shape=(2,2), dtype=np.float64)
+    _mrot = np.zeros(shape=(2, 2), dtype=np.float64)
     _mrot[0] = (np.cos(_theta), np.sin(_theta))
     _mrot[1] = (-np.sin(_theta), np.cos(_theta))
 
     return _mrot
 
-def countExtn(fimg, extname='SCI'):
-    """
-    Return the number of 'extname' extensions, defaulting to counting the
-    number of SCI extensions.
-    """
-
-    closefits = False
-    if isinstance(fimg, str):
-        fimg = pf.open(fimg)
-        closefits = True
-
-    n = 0
-    for e in fimg:
-        if 'extname' in e.header and e.header['extname'] == extname:
-            n += 1
-
-    if closefits:
-        fimg.close()
-
-    return n
 
 """
 
 Primary function for creating an astrometric reference catalog.
 
 """
-def create_astrometric_catalog(inputs, **pars):
+
+
+def create_astrometric_catalog(inputs, catalog="GAIADR2", output="ref_cat.ecsv",
+                               gaia_only=False, table_format="ascii.ecsv", existing_wcs=None):
```

Comment:
In general it's more clear to set the default values of the keywords arguments in the function signature, especially if the number of arguments is not that big, as opposed to using `**kwargs` and then popping default values in the code. This allows a user to use simple (i)python tools to find out the default values. It also allows tools like pycodestyle and flake8 to catch mismatch of arguments, which was an issue in at least two functions where the docstrings defined parameters which were not used in the function.